cmd/print_test.go

```diff
@@ -236,3 +236,60 @@ func Test_getNamespaceInCurrentContext(t *testing.T) {
 			actualNamespace, err, expectedNamespace, nil)
 	}
 }
+
+func Test_getProviderSpecificConf(t *testing.T) {
+	value1 := "value1"
+	value2 := "value2"
+	testCases := []struct {
+		name                 string
+		providerSpecificConf map[string]*string
+		providers            []string
+		expected             map[string]map[string]string
+	}{
+		{
+			name:                 "No provider specific configuration",
+			providerSpecificConf: make(map[string]*string),
+			providers:            []string{"provider"},
+			expected:             map[string]map[string]string{},
+		},
+		{
+			name:                 "Provider specific configuration matching provider in the list",
+			providerSpecificConf: map[string]*string{"provider-conf": &value1},
+			providers:            []string{"provider"},
+			expected: map[string]map[string]string{
+				"provider": {"conf": value1},
+			},
+		},
+		{
+			name: "Provider specific configuration matching providers in the list with multiple providers",
+			providerSpecificConf: map[string]*string{
+				"provider-a-conf1": &value1,
+				"provider-b-conf2": &value2,
+			},
+			providers: []string{"provider-a", "provider-b", "provider-c"},
+			expected: map[string]map[string]string{
+				"provider-a": {"conf1": value1},
+				"provider-b": {"conf2": value2},
+			},
+		},
+		{
+			name:                 "Provider specific configuration not matching provider in the list",
```

Comment:
I liked that you added this test. Should we `log.Warn` in if such case happen? (I mean in the library code, not in the test)

Reply:
For now and following the common implementation, I've added a simple `log.Printf("Warning: …")`, from the std lib. But I've seen providers (Istio, Kong) relying on [k8s.io/klog/v2](https://pkg.go.dev/k8s.io/klog/v2@v2.110.1). Do we want to make a call on this?

Reply:
> Do we want to make a call on this?
Sounds good to me